packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -120,6 +126,18 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 
 	#status: WebSocketShardStatus = WebSocketShardStatus.Idle;
 
+	private identifyCompressionEnabled = false;
+
+	/**
+	 * @privateRemarks
+	 *
+	 * This is needed because `this.strategy.options.compression` is not an actual reflection of the compression method
+	 * used, but rather the compression method that the user wants to use. This is because the libraries could just be missing.
+	 */
+	private get transportCompressionEnabled() {
+		return this.strategy.options.compression !== null && (Boolean(this.nativeInflate) || Boolean(this.zLibSyncInflate));
```

Comment:
Single `Boolean` call:
```suggestion
		return this.strategy.options.compression !== null && Boolean(this.nativeInflate || this.zLibSyncInflate);
```

Alternatively:
```suggestion
		return this.strategy.options.compression !== null && (this.nativeInflate ?? this.zLibSyncInflate) !== null;
```